simple-factsheet.Rmd

```diff
@@ -23,91 +28,401 @@ library(tidyverse)
+                                       temp_working_diff >= 6 ~ "higher for children of immigrants than",
+                                       temp_working_diff > -6 & temp_working_diff <= -3 ~ "slightly lower for children of immigrants than",
+                                       temp_working_diff <= -6 ~ "lower for children of immigrants than"), 
+         
+         text_5_lowinc = case_when(temp_working_diff <= -3 | (share_above200povr_coi >- share_above200povr_nat) ~ "Around",
+                                   temp_working_diff > -3 ~ "Yet,",
+                                   TRUE ~ NA_character_),
+         
+         text_6_race = case_when((share_asian_coi >= 75 |
+                                  share_black_coi >= 75 | 
+                                  share_hispanic_coi >= 75 |
+                                  share_white_coi >= 75 ~ " fairly "),
+                                 TRUE ~ " "),
+         
+         temp_1_uninsure = (share_noinsure_all_0809) - (share_noinsure_all_1617),
+         
+         text_1_uninsure = case_when(temp_1_uninsure >= 1 ~ "fallen",
+                                     abs(temp_1_uninsure) < 1 ~ "not changed dramatically",
+                                     temp_1_uninsure <= 1 ~ "risen",
+                                     TRUE ~ NA_character_)) %>%
+  select(-starts_with("temp"))
+  
+# Dataset for race paragraph 
+# handling oth/mult race separately since it doesn't move around like the other racial/ethnic groupings
+race_subset <- states_subset %>%
+ select(c("StateName","share_asian_coi","share_black_coi","share_hispanic_coi","share_white_coi",
+        race_COIpop_growmost_0506to1718)) %>%
+  pivot_longer(cols = starts_with("share"),
+              names_to = "race",
+              values_to = "share") %>%
+  mutate(race = case_when(str_detect(race,"asian") ~ "Asian",
+                          str_detect(race,"black") ~ "Black",
+                          str_detect(race,"hispanic") ~ "Hispanic",
+                          str_detect(race,"white") ~ "white")) %>%
+  rename(fastest = race_COIpop_growmost_0506to1718) %>%
+  mutate(fastest = ifelse(tolower(race) == fastest,1,0)) %>%
+  arrange(-share) %>%
+  mutate(rank = 1:nrow(.)) 
+
+  # Ordered vectors of percents, labels for race groups, and text strings (mostly empty) to insert in race paragraph 
+  race_percents <- race_subset$share
+  race_labels <- race_subset$race
+  race_growth <- ifelse(race_subset$fastest == 1, "—the fastest-growing group","")
+
+  urban_blue <- urbnthemes::palette_urbn_diverging[7]
+  urban_black <- urbnthemes::palette_urbn$categorical[[3]][3]
+  urban_grey <- urbnthemes::palette_urbn$categorical[[4]][4]
+  urban_lightblue <- urbnthemes::palette_urbn$sequential[[2]][1]
+  
+## Creating Figure 1
+  
+  fig1_data <- states_subset %>%
+    select(c("share_natpar","share_uscitz_coi","share_noncitz_coi","share_unkpar")) %>%
+    pivot_longer(everything()) %>%
```

Comment:
I don't think `everything()` is necessary.

Reply:
doesn't run for me without `everything()`! I get `Error: `cols` must select at least one column.`